.contrib/Parser/DATAS/02 - Outdoor Zones/08 Broken Isles/2 - Quest Chains.lua

```diff
@@ -95,101 +95,159 @@ _.Zones =
-						q(43531),	-- Into the Nighthold
-						q(43530),	-- The Nighthold: Delusions of Grandeur
-						q(43532),	-- The Nighthold: Darkness Calls
-						q(43533, {	-- Balance of Power
-							cl(1, {
-							  ["g"] = {
-								artifact(805),
-								artifact(669),
-								artifact(164),
-							  }
-							  }),
-							cl(2, {
-							  ["g"] = {
-								artifact(16),
-								artifact(545),
-								artifact(856),
-							  }
-							}),
-							cl(3, {
-							  ["g"] = {
-								artifact(219),
-								artifact(462),
-								artifact(481),
-							  }
-							}),
-							cl(4, {
-							  ["g"] = {
-								artifact(237),
-								artifact(765),
-								artifact(71),
-							  }
-							}),
-							cl(5, {
-							  ["g"] = {
-								artifact(735),
-								artifact(753),
-								artifact(255),
-							  }
-							}),
-							cl(6, {
-							  ["g"] = {
-								artifact(368),
-								artifact(371),
-								artifact(403),
-							  }
-							}),
-							cl(7, {
-							  ["g"] = {
-								artifact(310),
-								artifact(682),
-								artifact(781),
-							  }
-							}),
-							cl(8, {
-							  ["g"] = {
-								artifact(184),
-								artifact(499),
-								artifact(134),
-							  }
-							}),
-							cl(9, {
-							  ["g"] = {
-								artifact(198),
-								artifact(329),
-								artifact(818),
-							  }
-							}),
-							cl(10, {
-							  ["g"] = {
-								artifact(517),
-								artifact(348),
-								artifact(525),
-							  }
-							}),
-							cl(11, {
-							  ["g"] = {
-								artifact(419),
-								artifact(54),
-								artifact(434),
-								artifact(277),
-							  }
-							}),
-							cl(12, {
-							  ["g"] = {
-								artifact(36),
-								artifact(563),
-							  }
-							}),
+						q(43530, {	-- The Nighthold: Delusions of Grandeur
+							["sourceQuest"] = 43898,	-- Preparing to Move
+							["provider"] = { "n", 101083 },	-- Thalrenus Rivertree
+							["coord"] = { 37.8, 47.3, 680 },
+							["maps"] = { 680 },	-- Suramar					
+						}),
+						q(43531, {	-- Into the Nighthold
+							["sourceQuest"] = 43898,	-- Preparing to Move
+							["provider"] = { "n", 111826 },	-- Archmage Kalec
+							["coord"] = { 37.8, 47.4, 680 },
+							["maps"] = { 680 },	-- Suramar					
+						}),
+						q(43532, {	-- The Nighthold: Darkness Calls
+							["sourceQuest"] = 43898,	-- Preparing to Move
+							["provider"] = { "n", 111826 },	-- Archmage Kalec
+							["coord"] = { 37.8, 47.4, 680 },
+							["maps"] = { 680 },	-- Suramar					
+						}),
+						q(43533, { 	-- Balance of Power
+							["sourceQuests"] = {
+												43530,	-- The Nighthold: Delusions of Grandeur
+												43531,	-- Into the Nighthold
+												43532,	-- The Nighthold: Darkness Calls
+												},
+							["provider"] = { "n", 111826 },	-- Archmage Kalec
+							["coord"] = { 37.8, 47.4, 680 },
+							["maps"] = { 680 },	-- Suramar		
+							["g"] = {	
+								cl(1, {
```

Comment:
While at it consider replacing numbers with named constants for classes for better readability (1 would become WARRIOR, 2 - PALADIN, etc, full list is in `_main.lua`)